StoreKitUnitTests/StoreProductTests.swift

```diff
@@ -261,4 +267,35 @@ class StoreProductTests: StoreKitConfigTestCase {
         testSession.locale = Locale(identifier: "es_ES")
     }
 
+    private func expectEqualProducts(_ productA: StoreProductType, _ productB: StoreProductType) {
+        expect(productA.productIdentifier) == productB.productIdentifier
+        expect(productA.localizedDescription) == productB.localizedDescription
+        expect(productA.price) == productB.price
+        expect(productA.localizedPriceString) == productB.localizedPriceString
+        expect(productA.productIdentifier) == productB.productIdentifier
+        expect(productA.isFamilyShareable) == productB.isFamilyShareable
+        expect(productA.localizedTitle) == productB.localizedTitle
+
+        expect(productA.isFamilyShareable) == productB.isFamilyShareable
+
+        expect(productA.discounts) == productB.discounts
+
+        if productA.subscriptionPeriod == nil {
+            expect(productB.subscriptionPeriod).to(beNil())
+        } else {
+            expect(productA.subscriptionPeriod) == productB.subscriptionPeriod
+        }
+
+        if productA.introductoryPrice == nil {
```

Comment:
I've added a few more checks to these tests too for the new properties added in #1038.